tests/integration/records/test_api_literature.py

```diff
@@ -593,3 +593,10 @@ def test_add_and_remove_figs_and_docs_when_files_flag_disabled(
         record["figures"]
 
     assert record["documents"][0]["key"] == expected_doc_filename
+
+
+def test_create_record_from_db_depending_on_its_pid_type(base_app, db):
```

Comment:
what about tests for ``create_or_update``?

Reply:
create_or_update is using either the get_record which does the "class resolution" or the create one so by testing those I am testing that it is using the proper class. But if you think it would be better to have it more explicitly testing, then I can add that as well